manifests/init.pp

```diff
@@ -591,6 +591,16 @@
   -> Class['::sensu::server::service']
   -> Class['::sensu::enterprise::dashboard']
 
+  # Dependencies for Plugins #
+  # Without this line, plugins installed with the sensu_gem provider may get
+  # evaluated before the sensu package, which, on sensu package upgrades that
+  # install a new version of Ruby, would mean that two runs of Puppet would be
+  # necessary to get the plugins installed into the new embedded Ruby folder.
+  Package['sensu']
+  ~> Package<| provider == 'sensu_gem' |>
+  ~> Service<| title == 'sensu-client' or title == 'sensu-server' or title == 'sensu-api'|>
+  #~> Service['sensu-client', 'sensu-server', 'sensu-api']
```

Comment:
Still working through the enterprise side?

These will need the corresponding spec tests.

Reply:
I need to pull out that comment for sure.  As for the enterprise side, I'm waiting for @agoddard to reply to what services (if any) on the enterprise we will need to bump (see: https://github.com/sensu/sensu-puppet/issues/542#issuecomment-331249697).  I don't think we need to bounce that service, but I'll add it when I get a reply